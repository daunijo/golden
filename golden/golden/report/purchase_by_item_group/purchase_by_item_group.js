// Copyright (c) 2016, RSS and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Purchase By Item Group"] = {
	"filters": [
		{
			"fieldname":"purchase_budget",
			"label": __("Purchase Budget"),
			"fieldtype": "Link",
			"options": "Purchase Budget",
			"reqd": 1,
			"get_query": function() {
				return {
					"doctype": "Purchase Budget",
					"filters": {
						"docstatus": 1,
					}
				}
			}
		},
	]
}
