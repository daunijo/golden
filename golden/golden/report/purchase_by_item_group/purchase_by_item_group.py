# Copyright (c) 2013, RSS and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.utils import flt, getdate

def execute(filters=None):
	columns = get_columns()
	data = []

	conditions = get_conditions(filters)
	# bulk_po = frappe.db.get_value("Purchase Budget", frappe.db.escape(filters["purchase_budget"]), "bulk_po")
	# frappe.throw(_("select distinct(i.item_group) from `tabPurchase Order Item` poi inner join `tabItem` i on poi.item_code = i.item_code inner join `tabItem Group` ig on i.item_group = ig.`name` where poi.parent in ({0})").format(bulk_po))
	sl_entries = frappe.db.sql("select distinct(item_group), parent from `tabPurchase Budget Item` where %s order by item_group asc" % conditions, as_dict=1)
	for cl in sl_entries:
		total = frappe.db.sql("select sum(net_amount) from `tabPurchase Budget Item` where item_group = %s and parent = %s", (cl.item_group, cl.parent))[0][0]
		total2 = frappe.db.sql("select sum(net_amount) from `tabPurchase Budget Item` where parent = %s", cl.parent)[0][0]
		percen = flt(total) / flt(total2) * 100
		data.append([cl.item_group, total, percen])

	return columns, data

def get_columns():
	"""return columns"""

	columns = [
		_("Item Group")+":Link/Item Group:200",
		_("Total")+":Float:120",
		_("Kontribusi")+":Percent:100",
	]

	return columns

def get_conditions(filters):
	conditions = ""
	if filters.get("purchase_budget"):
		conditions += " parent = '%s'" % frappe.db.escape(filters["purchase_budget"])
	return conditions
