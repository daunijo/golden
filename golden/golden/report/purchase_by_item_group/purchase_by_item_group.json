{
 "add_total_row": 0, 
 "apply_user_permissions": 1, 
 "creation": "2018-05-07 11:19:39.192485", 
 "disabled": 0, 
 "docstatus": 0, 
 "doctype": "Report", 
 "idx": 0, 
 "is_standard": "Yes", 
 "letter_head": "Standard", 
 "modified": "2018-05-07 11:20:03.512491", 
 "modified_by": "Administrator", 
 "module": "Golden", 
 "name": "Purchase By Item Group", 
 "owner": "Administrator", 
 "ref_doctype": "Item Group", 
 "report_name": "Purchase By Item Group", 
 "report_type": "Script Report", 
 "roles": [
  {
   "role": "Administrator"
  }, 
  {
   "role": "All"
  }
 ]
}